src/orquestra/quantum/measurements/parities.py

```diff
@@ -0,0 +1,198 @@
+        values = convert_dict_to_array(data["values"])
+        if data.get("correlations"):
+            correlations: Optional[List] = [
+                convert_dict_to_array(arr) for arr in data["correlations"]
+            ]
+        else:
+            correlations = None
+        return cls(values, correlations)
+
+
+def save_parities(parities: Parities, filename: AnyPath) -> None:
+    """Save parities to a file.
+
+    Args:
+        parities (orquestra.quantum.measurement.Parities): the parities
+        file (str or file-like object): the name of the file, or a file-like object
+    """
+    data = parities.to_dict()
+
+    with open(filename, "w") as f:
+        f.write(json.dumps(data, indent=2))
+
+
+def load_parities(file: LoadSource) -> Parities:
+    """Load parities from a file.
+
+    Args:
+        file (str or file-like object): the name of the file, or a file-like object.
+
+    Returns:
+        orquestra.quantum.measurement.Parities: the parities
+    """
+
+    with ensure_open(file) as f:
+        data = json.load(f)
+
+    return Parities.from_dict(data)
+
+
+def check_parity(
+    bitstring: Union[str, Sequence[int]], marked_qubits: Iterable[int]
+) -> bool:
+    """Determine if the marked qubits have even parity for the given bitstring.
+
+    Args:
+        bitstring: The bitstring, either as a tuple or little endian string.
+        marked_qubits: The qubits whose parity is to be determined.
+
+    Returns:
+        True if an even number of the marked qubits are in the 1 state, False
+            otherwise.
+    """
+    result = True
+    for qubit_index in marked_qubits:
+        if bitstring[qubit_index] == "1" or bitstring[qubit_index] == 1:
+            result = not result
+    return result
+
+
+def check_parity_of_vector(
+    bitstrings_vector: np.ndarray, marked_qubits: Iterable[int]
+) -> np.ndarray:
+    """Determine if the marked qubits have even parity for each bitstring in the given
+        vector.
+        NOTE: This performs the same functionality as `check_parity` but is much
+        faster as it uses vectorization to find the parity of multiple bitstrings
+        at once.
+
+    Args:
+        bitstring: A 2d array of bitstrings whose size is number of bistrings * number
+            of qubits
+        marked_qubits: The qubits whose parity is to be determined.
+
+    Returns:
+        A 1d array with size equal to number of bitstrings. Each entry is 1 if an even
+        number of the marked qubits of the corresponding bitstring are in the 1 state
+        and 0 if otherwise.
+    """
+    if not marked_qubits:
+        return np.ones(bitstrings_vector.shape[0])
+
+    # Check if an even number of the marked qubits of each bitstring are in the 1 state
+    bitstring_subset = bitstrings_vector[:, np.fromiter(marked_qubits, dtype=int)]
+    return (bitstring_subset.sum(axis=1) + 1) % 2
+
+
+def get_parities_from_measurements(
+    measurements: List[Tuple[int]], ising_operator
```

Comment:
nitpick (non-blocking): Can you type this `ising_operator` real quick?

Reply:
added it in the new commit.